From glikely@secretlab.ca Tue May  6 07:41:49 2008
From: Grant Likely <grant.likely@secretlab.ca>
Date: Tue, 6 May 2008 08:41:44 -0600
Subject: POWERPC: mpc5200: Fix unterminated of_device_id table
To: "Greg KH" <greg@kroah.com>
Message-ID: <fa686aa40805060741vd4befa4i1ea5ab6df4d54d9b@mail.gmail.com>
Content-Disposition: inline


From: Grant Likely <grant.likely@secretlab.ca>

commit bc775eac63c16dbcfabc4c6e949c0228edf3e11f upstream

If CONFIG_PPC_MPC5121 is not set, then the of_device_id table for the
mpc5200 serial driver will not get terminated with a NULL entry.

Signed-off-by: Grant Likely <grant.likely@secretlab.ca>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/serial/mpc52xx_uart.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/serial/mpc52xx_uart.c
+++ b/drivers/serial/mpc52xx_uart.c
@@ -1221,8 +1221,8 @@ static struct of_device_id mpc52xx_uart_
 #endif
 #ifdef CONFIG_PPC_MPC512x
 	{ .compatible = "fsl,mpc5121-psc-uart", .data = &mpc512x_psc_ops, },
-	{},
 #endif
+	{},
 };
 
 static int __devinit
